core/akka-runtime/src/test/scala/com/ing/baker/runtime/akka/actor/process_instance/ProcessInstanceSpec.scala

```diff
@@ -664,5 +676,284 @@ class ProcessInstanceSpec extends AkkaTestBase("ProcessInstanceSpec") with Scala
+
+      val interactionTransition: InteractionTransition = InteractionTransition(
+        eventsToFire, eventsToFire,
+        Seq.empty,
+        "Name",
+        "Name",
+        Map.empty,
+        Option.empty,
+        failureStrategy, Map.empty, false)
+
+      val logMock = mock[DiagnosticLoggingAdapter]
+
+      val output: String = ProcessInstance.getOutputEventName(interactionTransition, logMock)
+
+      assert(output == eventName)
+    }
+
+    "Should correctly determine the OutputEventName for Delayed Transitions if RetryWithIncrementalBackoff is used without FireEvent" in new TestSequenceNet {
+      override val sequence = Seq(
+        transition() { _ => Added(1) })
+
+      val eventName = "originalEvent1"
+      val exhaustedEvent = "exhaustedEvent"
+
+      val eventsToFire: Seq[EventDescriptor] = Seq(
+        EventDescriptor(eventName, Seq.empty)
+      )
+
+      val failureStrategy: InteractionFailureStrategy =
+        RetryWithIncrementalBackoff(Zero, 1.0, 1, Option.empty, Option.empty)
+
+      val interactionTransition: InteractionTransition = InteractionTransition(
+        eventsToFire, eventsToFire,
+        Seq.empty,
+        "Name",
+        "Name",
+        Map.empty,
+        Option.empty,
+        failureStrategy, Map.empty, false)
+
+      val logMock = mock[DiagnosticLoggingAdapter]
+
+      val output: String = ProcessInstance.getOutputEventName(interactionTransition, logMock)
+
+      assert(output == eventName)
+    }
+
+    "Should get correct wait time from state" in new TestSequenceNet {
+      override val sequence = Seq(
+        transition() { _ => Added(1) })
+
+      val eventName = "originalEvent1"
+
+      val eventsToFire: Seq[EventDescriptor] = Seq(
+        EventDescriptor(eventName, Seq.empty)
+      )
+
+      val interactionTransition: InteractionTransition = InteractionTransition(
+        eventsToFire, eventsToFire,
+        Seq(IngredientDescriptor("waitTime", types.RecordType(Seq(RecordField("seconds", Int64), RecordField("nanos", Int32))))),
+        "Name",
+        "Name",
+        Map.empty,
+        Option.empty,
+        BlockInteraction, Map.empty, false)
+
+      val duration: Duration = Duration.ofMillis(60000L)
+      val value: Value = Converters.toValue(duration)
+
+      val ingredients = Map[String, Value]("waitTime" -> value)
+
+      val output: Long = ProcessInstance.getWaitTimeInMillis(interactionTransition, RecipeInstanceState("id", "id", ingredients, Seq.empty))
+
+      assert(output == 60000L)
+    }
+
+    "Should get correct wait time from the predefined ingredients" in new TestSequenceNet {
+      override val sequence = Seq(
+        transition() { _ => Added(1) })
+
+      val eventName = "originalEvent1"
+
+      val eventsToFire: Seq[EventDescriptor] = Seq(
+        EventDescriptor(eventName, Seq.empty)
+      )
+
+      val duration: Duration = Duration.ofMillis(60000L)
```

Comment:
maybe we should use more descriptive names for duration and corresponding value, to make a clear distinciton between predefined ingredient and recipe ingredient. (In all the test to have a similar approach although not both are used in all tests)

Reply:
Instead of renaming I removed them, there is no need to make them a val since they are not re-used. Instead I directly initialise them in the ingredients map.